### PR TITLE
LRQA-74555 | master

### PR DESCRIPTION
**Pull request by weller0246**
https://issues.liferay.com/browse/LRQA-74555

### CODE REVIEW

**Comment by weller0246** [comment]
ci:test:sf

**Comment by weller0246** [comment]
ci:test:sf

File: portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/security/ldap/LDAP.testcase [hunk 1]
```diff
@@ -161,6 +161,67 @@ definition {
...
+
+			LDAP.updateLDAPDatabaseWithLDIFFile(ldifName = "addUser");
+
+			LDAP.updateLDAPDatabaseWithLDIFFile(ldifName = "addGroup");
+		}
+
+		task ("And: Manually triggers the LDAP import flow") {
+			Clustering.runGroovyScript(
+				nodePort = "8080",
+				scriptFile = "groovy-script-invoke-LDAPUSER-importon-master.groovy");
```
Comment (gaborlovasdotcom) [inline portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/security/ldap/LDAP.testcase:203]: Don't forget to use the new name for his script, based on BChan's formatting.

File: portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/security/ldap/LDAP.testcase [hunk 2]
```diff
@@ -161,6 +161,67 @@ definition {
...
+		}
+
+		task ("And: Removes the imported user on LDAP") {
+			LDAP.updateLDAPDatabaseWithLDIFFile(ldifName = "removeUser");
+		}
+
+		task ("And: Manually trigger the LDAP import flow") {
+			Clustering.runGroovyScript(
+				nodePort = "8080",
+				scriptFile = "groovy-script-invoke-LDAPUSER-importon-master.groovy");
```
Comment (gaborlovasdotcom) [inline portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/security/ldap/LDAP.testcase:215]: Same as above

**Comment by weller0246** [comment]
ci:test:sf

**Comment by weller0246** [comment]
ci:test:security:nocompile

File: portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/security/ldap/LDAP.testcase [hunk 3]
```diff
@@ -161,6 +161,67 @@ definition {
...
+
+		task ("When: User adds 2 user and to a group on LDAP, from wich only one have the mandatory fields to be imported") {
+			LDAP.installLDAPCommandUtil();
+
+			LDAP.updateLDAPDatabaseWithLDIFFile(ldifName = "addUser");
+
+			LDAP.updateLDAPDatabaseWithLDIFFile(ldifName = "addGroup");
+		}
+
+		task ("And: Manually triggers the LDAP import flow") {
```
[the remark this answers is not in the document]
Comment (gaborlovasdotcom) [reply portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/security/ldap/LDAP.testcase]: I agree it's more controlled and saves time for our tests!

**Review by gaborlovasdotcom — APPROVED** [review]

File: portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/security/ldap/LDAP.testcase [hunk 4]
```diff
@@ -97,6 +97,67 @@ definition {
...
+	test CanRemoveUserFromGroupThroughLDAP {
+		property apacheds.enabled = "true";
+		property test.name.skip.portal.instance = "LDAP#CanRemoveUserFromGroupThroughLDAP";
+
+		task ("Given: User enable DEBUG log level for LDAP related loggers") {
+			LDAP.enableLDAPDebugLogs();
+		}
+
+		task ("And: Add LDAP server and test connection") {
+			PortalSettings.addLDAPServer();
```
Comment (JoyceWang08211) [inline portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/security/ldap/LDAP.testcase:111]: Replace simple commands with LDAP.addLDAPServerAndTestConnection();

File: portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/security/ldap/LDAP.testcase [hunk 5]
```diff
@@ -97,6 +97,67 @@ definition {
...
+			var enableSettingList = "enabled,ldapImportEnabled,importOnStartup";
+
+			LDAP.enableLDAPSettings(enableSettingList = "${enableSettingList}");
+
+			PortalSettings.changeGroupImportMethod(groupImportMethod = "Group");
+
+			PortalSettings.setImportIntervalTime(importIntervalTime = "1");
+		}
+
+		task ("When: User adds 2 user and to a group on LDAP, from wich only one have the mandatory fields to be imported") {
```
Comment (JoyceWang08211) [inline portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/security/ldap/LDAP.testcase:128]: "from **which**"

**Comment by JoyceWang08211** [comment]
I think no need to introduce many And clauses. To make the prerequisites and the expected test results more clearer, we can improve the tasks as follows:

            task ("Given: User enable DEBUG log level for LDAP related loggers") {

		}

		task ("And: Add LDAP server, test connection and configure LDAP settings") {
			
		}

		task ("When: Add 2 users and 1 usergroup on LDAP and manually trigger the LDAP import flow") {
		}

                task ("Then: Only one user which contains the mandatory fields will be imported") {
		}

                task ("When: Remove the imported user on LDAP and trigger the LDAP import flow again") {
		}

		task ("Then: The imported group should be empty and the user should be unassigned") {
		}
```

**Comment by JoyceWang08211** [comment]
@weller0246 Nice macrodefs! Well done.

**Comment by weller0246** [comment]
ci:test:sf

**Comment by weller0246** [comment]
LDIF files are sorted and tasks are shortened down to one **AND** per section. 

cc @gaborlovasdotcom  @JoyceWang08211

**Comment by weller0246** [comment]
ci:test:security:nocompile

**Review by gaborlovasdotcom — APPROVED** [review]
LGTM

**Comment by gaborlovasdotcom** [comment]
ci:test:sf

**Comment by gaborlovasdotcom** [comment]
ci:forward

**Comment by gaborlovasdotcom** [comment]
ci:test:ldap:nocompile

**Comment by weller0246** [comment]
In the last ldif file **removeUser** a wrong DN was used for the users identification. I corrected it, starting a re-run
@gaborlovasdotcom

**Comment by weller0246** [comment]
ci:test:ldap:nocompile

**Comment by weller0246** [comment]
ci:stop

**Comment by weller0246** [comment]
ci:test:ldap:nocompile

**Comment by weller0246** [comment]
ci:test:ldap:nocompile

**Comment by weller0246** [comment]
ci:test:ldap:nocompile

**Comment by weller0246** [comment]
Test is failing because [LPS-153043](https://issues.liferay.com/browse/LPS-153043) is blocking it. If you have no more comments we can forward it.

**Comment by gaborlovasdotcom** [comment]
ci:forward

**Pull request closed without merging**